### ISSUE TITLE
attention_x and attention_y give wrong values in some cases when image is resized

### ISSUE THREAD

**Issue by ejoebstl** [lovell/sharp#3675]
This bug report is regarding the `attention_x` and `attention_y` output params introduced in #3470.

The output given for `attention_x` and `attention_y` is incorrect when the image is resized. My assumption is that two-stage resizing is not handled correctly, specifically [here](https://github.com/lovell/sharp/blob/main/src/pipeline.cc#L484).

I find the resizing code in `pipeline.cc` very hard to follow - @lovell do you have any advice on this by chance?

**Comment by lovell** [comment]
Please can you provide a sample image and code that allows someone else to reproduce.
